internal/api/backresthandler.go

```diff
@@ -268,6 +298,107 @@ func (s *BackrestHandler) RemoveRepo(ctx context.Context, req *connect.Request[t
 	return connect.NewResponse(cfg), nil
 }
 
+// addSftpHostKey parses an SFTP URI and adds the host's public key to the user's known_hosts file.
+// This is equivalent to what `ssh-keyscan` does.
+func (s *BackrestHandler) addSftpHostKey(uri string) error {
```

Comment:
let's factor out this functionality into an sftputil package perhaps e.g. internal/api/sftputil so that it's easier to hit with test coverage. 

I think keyscan might be a bit hard to test, but key generation etc we definitely can.